### PR TITLE
BUG: don't use array newton unless size is greater than 1

### LINKED ISSUE #9608: Version 1.2.0 introduces `too many indices for array` error in `optimize.newton()`
Since version 1.2.0, `optimize.newton` fails when passing an array with a single entry and when passing `fprime2`

### Reproducing code example:

```python
from scipy.optimize import newton
import numpy as np

def f(x):
    return x**2

def fprime(x):
    return 2 * x

def fprime2(x):
    return 2

def test():
    x0 = np.array([-2], dtype=np.float32)

    optimum = newton(f, x0=x0)  # OK
    print(optimum)
    optimum = newton(f, x0=x0, fprime=fprime)  # OK
    print(optimum)
    optimum = newton(f, x0=x0, fprime=fprime, fprime2=fprime2)  # Error
    print(optimum)

test()
```

### Error message:

```
[-1.95439407e-08]
[-7.4505806e-09]
Traceback (most recent call last):
  File "lolx.py", line 23, in <module>
    test()
  File "lolx.py", line 20, in test
    optimum = newton(f, x0=x0, fprime=fprime, fprime2=fprime2)  # Error
  File "/home/nico/.virtualenvs/pygbm/lib/python3.7/site-packages/scipy/optimize/zeros.py", line 263, in newton
    full_output)
  File "/home/nico/.virtualenvs/pygbm/lib/python3.7/site-packages/scipy/optimize/zeros.py", line 380, in _array_newton
    dp = dp / (1.0 - 0.5 * dp * fder2[nz_der] / fder[nz_der])
IndexError: too many indices for array

```

### Scipy/Numpy/Python version information:

Works in 1.1.0
```
1.1.0 1.15.4 sys.version_info(major=3, minor=7, micro=1, releaselevel='final', serial=0)
```

Fails in 1.2.0
```
1.2.0 1.15.4 sys.version_info(major=3, minor=7, micro=1, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by mikofski**
closes #9608

### CODE REVIEW

**Comment by mikofski** [comment]
FYI: the [Azure plpeline check failure](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=61) is unrelated

File: scipy/optimize/zeros.py [hunk 1]
```diff
@@ -99,9 +99,10 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
     derivative `fprime2` of `func` is also provided, then Halley's method is
     used.
 
-    If `x0` is a sequence, then `newton` returns an array, and `func` must be
-    vectorized and return a sequence or array of the same shape as its first
-    argument.
+    If `x0` is a sequence with more than one item, then `newton` returns an
```
[the remark this answers is not in the document]
Comment (mikofski) [reply scipy/optimize/zeros.py]: Yes, that's it exactly.

**Comment by mikofski** [comment]
@NicolasHug can you test out this branch and see if it addresses your issue?

@rgommers can you review this?

Thanks and Happy Holidays! 🎅 🎄 🎁 ❄️ ⛄️ 🦌

File: scipy/optimize/tests/test_zeros.py [hunk 2]
```diff
@@ -619,3 +619,22 @@ def fpp(x):
...
+
+    def fp(x):
+        return 2 * x
+
+    def fpp(x):
+        return 2
+
+    x0 = np.array([-2], dtype=np.float32)
+    rt, r = newton(f, x0, fprime=fp, fprime2=fpp, full_output=True)
+    assert(r.converged)
```
Comment (andyfaff) [inline scipy/optimize/tests/test_zeros.py]: you need to use `np.testing.assert_` instead of a bare assert

Comment (larsoner) [reply scipy/optimize/tests/test_zeros.py]: I don't think that's the case now that we use pytest

**Comment by NicolasHug** [comment]
@mikofski Just checked, yes this PR fixes my original issue.

Thanks a lot!!

File: scipy/optimize/tests/test_zeros.py (hunk 2 above)
Comment (larsoner) [reply scipy/optimize/tests/test_zeros.py]: (see #8583)

**Comment by tylerjereddy** [comment]
Is this ready to merge? I'll backport this one it is.

**Comment by mikofski** [comment]
Done from my perspective, but @rgommers or @pvanmulbregt may have feedback?

File: scipy/optimize/tests/test_zeros.py (hunk 2 above)
Comment (pvanmulbregt) [reply scipy/optimize/tests/test_zeros.py]: Is there a test that newton fails for a mismatch between f and app when passed a longer array?  E.g. continuing the example above, something like 
```
x0 = np.array([-2, -3], dtype=np.float32)
# This next invocation should fail
rt, r = newton(f, x0, fprime=fp, fprime2=fpp, full_output=True)
```

Comment (mikofski) [reply scipy/optimize/tests/test_zeros.py]: I looked through `test_zeros.py` but there is no such test currently. Should I add a test to catch the exception?

Comment (pvanmulbregt) [reply scipy/optimize/tests/test_zeros.py]: It's now part of the contract in the docstring so adding a check on a mismatch would be reasonable.

**Comment by mikofski** [comment]
Ok @tylerjereddy I think it's ready now. Note the [Travis errors](https://travis-ci.org/scipy/scipy/jobs/476137655#L1246) are unrelated

**Review by pvanmulbregt — APPROVED** [review]

**Comment by tylerjereddy** [comment]
Ok, original issue reporter confirms issue is fixed, the added unit test code appears to match the originally reported issue, and seems core dev reviews / concerns addressed.

**Comment by tylerjereddy** [comment]
Thanks, @mikofski

**Pull request merged**